scripts/run-auto-pid-tuner

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Petter Reinholdtsen
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA.
+
+"""
+
+LinuxCNC automated PID auto-tuner
+=================================
+
+Automatically conduct a automatic PID tuning using the pid
+LinuxCNC component.
+
+This script first home the axis, then move it to the middle of its range.
+Also extend the f-error value used by the PID controller.  Next, disable
+the PID controller (tune-mode=1), start the tuning process (tune-start=1)
+and wait for it to finish before enabling the PID controller again.
+
+This script assume the pid component is configured like this:
+
+  loadrt pid names=pid.x,pid.y,pid.z
+
+It also assume the axis ranges are correct to be able to position the axes
+in their middle positions.
+"""
+
+import time
+
+import linuxcnc
+import linuxcnc_util
+import hal
+
+axis = 'x'
+axisid = 0
+
+# Some values that need to be replaced during tuning
+replace = {
+#    f"pid.{axis}.tune-cycles": "0x40",  # Set number of back and forth movement
+    f"pid.{axis}.tune-effort": "0.2",  # Set effort, should start low
```

Comment:
Why `tune-cycles` is commented and in hex?
Is `tune-effort` in PID output pin units or something else?